### PR TITLE
Add radial menu

### PR DESCRIPTION
**Pull request by jellysnake**
Added a new radial menu as per issue #2713.

This needs more testing done on it before it should be merged. An example and documented .ui orefab is [here](https://hastebin.com/lepawigope.json).

Todo:

- [x] Possibly change to having sections as widgets themselves
- [ ] Add support for sub menus
- [x] Add support for other stuff to happen on a selection pressed. Ie, a butten press or event fire.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/rendering/nui/widgets/UIRadialRing.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        return region.size();
+    }
+
+    private int getSectionOver(Vector2i mousePos) {
+        logger.info(mousePos.toString());
+        mousePos.x -= radius;
+        mousePos.y -= radius;
+
+        logger.info(mousePos.toString());
+        double angle = Math.atan2(mousePos.y(), mousePos.y());        logger.info(angle + " ang");
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/widgets/UIRadialRing.java]: Might have accidentally lost a newline here?

File: engine/src/main/java/org/terasology/rendering/nui/widgets/UIRadialEmpty.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.widgets;
+
+import org.terasology.math.geom.Rect2i;
+import org.terasology.rendering.assets.texture.TextureRegion;
+import org.terasology.rendering.nui.Canvas;
+
+public class UIRadialEmpty extends UIRadialSection {
```
Comment (Cervator) [inline engine/src/main/java/org/terasology/rendering/nui/widgets/UIRadialEmpty.java]: Would be great with class-level Javadoc on this and similar :-)

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
Linking to request https://github.com/Terasology/TutorialNui/issues/3 for example sometime :-)

**Pull request merged**